src/eventloop.jl

```diff
@@ -1,17 +1,28 @@
 """
-    eventloop(socket)
+    eventloop(socket, kernel)
 
 Generic event loop for one of the [kernel
 sockets](https://jupyter-client.readthedocs.io/en/latest/messaging.html#introduction).
 """
-function eventloop(socket)
+function eventloop(socket, kernel)
     task_local_storage(:IJulia_task, "write task")
     try
         while true
-            msg = recv_ipython(socket)
+            local msg
             try
-                send_status("busy", msg)
-                invokelatest(get(handlers, msg.header["msg_type"], unknown_request), socket, msg)
+                msg = recv_ipython(socket, kernel)
+            catch e
+                if isa(e, EOFError)
+                    # The socket was closed
+                    return
+                else
+                    rethrow()
+                end
+            end
+
+            try
+                send_status("busy", kernel, msg)
+                invokelatest(get(handlers, msg.header["msg_type"], unknown_request), socket, kernel, msg)
```

Comment:
This appears to hit `comm_open`, `comm_msg` and `comm_close`, but those don't have the same signature as the one used here

Reply:
Fixed in bb7cef4.